### PR TITLE
Adding curve checks to integratedTests

### PR DESCRIPTION
**Pull request by cssherman**

### CODE REVIEW

File: scripts/geos_ats_package/geos_ats/helpers/curve_check.py [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+        xd = np.zeros((len(tb), S))
+        for ii in range(S):
+            xd[:, ii] = interp1d(ta, xc[:, ii])(tb)
+
+        # Return the array to it's expected shape
+        N[0] = M
+        return np.reshape(xd, N)
+
+
+def evaluate_external_script(script, fn, data):
```
Comment (TotoGaz) [inline scripts/geos_ats_package/geos_ats/helpers/curve_check.py:50]: IIUC, the CLI string are converted into callable using python imports.
Would it make sense to convert this information into a callable asap, such that you manipulate a std python `__call__` without any further considerations?

**Review by TotoGaz — APPROVED** [review]

File: scripts/geos_ats_package/geos_ats/test_steps.py [hunk 2]
```diff
@@ -397,24 +397,27 @@ class geos(TestStepBase):
...
-        else:
-            self.checksteps = []
+        self.checksteps = []
+        if checkOption in ["all", "curvecheck"]:
+            if curvecheck_params is not None:
+                self.checksteps.append(curvecheck(curvecheck_params, **kw))
+
+        if checkOption in ["all", "restartcheck"]:
+            if restartcheck_params is not None:
+                self.checksteps.append(restartcheck(restartcheck_params, **kw))
```
[the remark this answers is not in the document]
Comment (cssherman) [reply scripts/geos_ats_package/geos_ats/test_steps.py:420]: Ahh, yes that makes sense.  At a minimum we should put out a log statement, since we are trying to enforce those for all tests moving forward.

File: tests/allTests/HydrofracturingSolver/Hydrofracturing.ats [hunk 3]
```diff
@@ -1,43 +1,186 @@
...
-             independent=True,
-             steps=(geos(deck="kgdEdgeBased_C3D6_smoke.xml",
-                         name="kgdEdgeBased_C3D6",
-                         np=N,
-                         ngpu=N,
-                         x_partitions=nx,
-                         y_partitions=ny,
-                         z_partitions=nz,
-                         restartcheck_params=restartcheck_params), ))
+curvecheck_params_base = {}
```
Comment (CusiniM) [inline tests/allTests/HydrofracturingSolver/Hydrofracturing.ats:7]: At some point it would be nice if we could create actual classes to contain all these parameters instead of using dictionaries. I know they are popular in python but I personally find them hard to follow and quite error prone since there is no static verification.

**Review by CusiniM — APPROVED** [review]
